src/runtime/pickle_runner.ts

```diff
@@ -395,9 +426,80 @@ export default class PickleRunner {
         },
       })
     }
-    return await this.invokeStep(
-      testStep.pickleStep,
-      testStep.stepDefinitions[0]
+    let stepResult
+    let afterStepHooksResult
+    const beforeStepHooksResult = await this.runStepHooks(
+      this.getBeforeStepHookDefinitions()
     )
+
+    if (beforeStepHooksResult.status !== Status.FAILED) {
+      stepResult = await this.invokeStep(
+        testStep.pickleStep,
+        testStep.stepDefinitions[0]
+      )
+      if (stepResult.status === Status.PASSED) {
```

Comment:
I think we would still want to run the `AfterStep` hooks if the step failed. A common cited use case for this feature is to take a screenshot or attach an API response if the step fails.